File: src/main/java/com/example/probation/service/UsersService.java
package com.example.probation.service;

import com.example.probation.core.entity.User;

import java.util.List;
import java.util.Optional;

public interface UsersService {
    User registerNewUser(final User newUser);

    void redefineRating(final User winner, final User loser);

    List<User> getUsersForComparison();

    List<User> getTopUsersByRating();

    Integer calculateWinnerRating(Integer currentRating);

    Integer calculateLoserRating(Integer currentRating);

    Optional<User> findByUserName(String username);

    Optional<User> getUserByToken(String token);

    User saveRegisteredUser(String token);

    User getCurrentUser();

    boolean checkEmailExistence(String email);

    boolean checkUsernameExistence(String email);
}

File: src/main/kotlin/com/example/probation/service/UsersService.kt
package com.example.probation.service

import com.example.probation.core.dto.ChangePasswordDto
import com.example.probation.core.entity.User

interface UsersService {
    fun registerNewUser(newUser: User): User

    fun redefineRating(winner: User, loser: User)

    fun getUsersForComparison(): List<User>

    fun getTopUsersByRating(): List<User>

    fun calculateWinnerRating(currentRating: Int): Int

    fun calculateLoserRating(currentRating: Int): Int

    fun findByUserName(username: String): User?

    fun getUserByToken(token: String): User?

    fun saveRegisteredUser(token: String): User

    fun getCurrentUser(): User

    fun checkEmailExistence(email: String): Boolean

    fun checkUsernameExistence(username: String): Boolean

    fun blockUser(userId: Long)

    fun unblockUser(userId: Long)

    fun deleteUser(userId: Long)

    fun changePersonalData(id: Long, user: User)

    fun changePassword(changePasswordDto: ChangePasswordDto)

    fun checkUniqueNewName(newName: String, id: Long): Boolean

    fun checkUniqueNewEmail(newEmail: String, id: Long): Boolean
}